pkg/registry/apis/provisioning/jobs/pullrequest/worker.go

```diff
@@ -64,82 +72,55 @@ func (c *PullRequestWorker) Process(ctx context.Context,
 		return errors.New("pull request job submitted targeting repository that is not a Reader")
 	}
 
-	parser, err := c.parsers.GetParser(ctx, reader)
-	if err != nil {
-		return fmt.Errorf("failed to get parser for %s: %w", repo.Config().Name, err)
-	}
-
 	logger := logging.FromContext(ctx).With("pr", options.PR)
 	logger.Info("process pull request")
 	defer logger.Info("pull request processed")
 
 	progress.SetMessage(ctx, "listing pull request files")
 	base := cfg.GitHub.Branch
-	ref := options.Hash
-	files, err := prRepo.CompareFiles(ctx, base, ref)
+	files, err := prRepo.CompareFiles(ctx, base, options.Ref)
 	if err != nil {
 		return fmt.Errorf("failed to list pull request files: %s", err.Error())
 	}
 
-	progress.SetMessage(ctx, "clearing pull request comments")
-	if err := prRepo.ClearAllPullRequestFileComments(ctx, options.PR); err != nil {
-		return fmt.Errorf("failed to clear pull request comments: %+v", err)
-	}
+	files = onlySupportedFiles(files)
 
 	if len(files) == 0 {
 		progress.SetFinalMessage(ctx, "no files to process")
 		return nil
 	}
 
-	if len(files) > 1 {
-		progress.SetFinalMessage(ctx, "too many files to preview")
-		return nil
-	}
-
-	f := files[0]
-	progress.SetMessage(ctx, "processing file preview")
-
-	if err := resources.IsPathSupported(f.Path); err != nil {
-		progress.SetFinalMessage(ctx, "file path is not supported")
-		return nil
-	}
-
-	fileInfo, err := prRepo.Read(ctx, f.Path, ref)
-	if err != nil {
-		return fmt.Errorf("read file: %w", err)
-	}
-
-	_, err = parser.Parse(ctx, fileInfo)
+	parser, err := c.parsers.GetParser(ctx, reader)
 	if err != nil {
-		if errors.Is(err, resources.ErrUnableToReadResourceBytes) {
-			progress.SetFinalMessage(ctx, "file changes is not valid resource")
-			return nil
-		} else {
-			return fmt.Errorf("parse resource: %w", err)
-		}
+		return fmt.Errorf("failed to get parser for %s: %w", repo.Config().Name, err)
 	}
 
-	// Preview should be the branch name if provided, otherwise use the commit hash
-	previewRef := options.Ref
-	if previewRef == "" {
-		previewRef = ref
+	var render ScreenshotRenderer
+	if repo.Config().Spec.GitHub.GenerateDashboardPreviews {
```

Comment:
This should be handle in the `ChangeEvaluator` which uses the renderer.

Reply:
then it would be the only thing that requires knowlege of the repo spec -- and is really a boolean flag to use or not use the generator  (none of the mocks currently have to make a git repo config -- that seems nice)